scripts/generate_markdown_files.py

```diff
@@ -0,0 +1,143 @@
+        md_content += '<img src="..\images\oops_loki.png" width="500" height="400" title="Oops Loki">\n'
+    else:
+        for api_object_name, api_object in service_class.api_objects.items():
+            md_content += f"### {api_object.name}\n"
+            md_content += f"- **Type**: {api_object.type}\n"
+            md_content += f"- **Pattern**: `{api_object.pattern}`\n"
+            if api_object.min_length is not None:
+                md_content += f"- **Min Length**: {api_object.min_length}\n"
+            if api_object.max_length is not None:
+                md_content += f"- **Max Length**: {api_object.max_length}\n"
+            md_content += "\n"
+
+    return md_content
+
+
+# Function to find all subclasses of Service and generate markdown files
+def generate_markdown_files(output_dir):
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
+    service_classes = [cls for _, cls in getmembers(class_definitions, isclass) if
+                       issubclass(cls, class_definitions.Service) and cls is not class_definitions.Service]
+
+    print(f"Detected {len(service_classes)} service classes.")
+
+    updates_made = False
+    updated_classes = []
+
+    for service_class in service_classes:
+        instance = service_class()
+        if isinstance(instance.service_name_, str) and instance.service_name_:
+            md_content = create_markdown_for_class(instance)
+            file_name = f"{service_class.__name__.lower()}.md"
+            file_path = os.path.join(output_dir, file_name)
+            if os.path.exists(file_path):
+                with open(file_path, 'r', encoding='utf-8') as md_file:
+                    current_content = md_file.read()
+                if current_content == md_content:
+                    continue
+                else:
+                    update_type = 'updated'
+            else:
+                update_type = 'created'
+            with open(file_path, 'w', encoding='utf-8') as md_file:
+                md_file.write(md_content)
+            updates_made = True
+            updated_classes.append((service_class.__name__, update_type))
+
+    return updates_made, updated_classes
+
+
+# Function to update only the 'API Reference' section in mkdocs.yml
+def update_mkdocs_yml(mkdocs_yml_path, api_dir):
+    with open(mkdocs_yml_path, 'r', encoding='utf-8') as yml_file:
+        mkdocs_config = yaml.safe_load(yml_file)
+
+    # Generate the API Reference nav entries
+    api_files = sorted(os.listdir(api_dir))
+    api_nav = [{'{}'.format(os.path.splitext(f)[0].capitalize()): os.path.join('api', f).replace("\\", "/")} for f in
+               api_files]
+
+    # Update only the 'API Reference' section in the nav
+    nav_updated = False
+    original_nav = None
+    for item in mkdocs_config['nav']:
+        if isinstance(item, dict) and 'API Reference' in item:
+            original_nav = item['API Reference']
```

Comment:
there is a mix of logics here. you need to extract the logic that finds `item['API Reference']`. Then you check if it's `None` .  if it is - you need to create a new "API Reference" key, like you do in line 115.
otherwise - if it's different the the current `api_nav`, you need to update it.

This logic is way simpler.

Reply:
This will be addressed in a separate PR.